examples/apps/color_finder.py

```diff
@@ -0,0 +1,659 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2017 Anki, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License in the file LICENSE.txt or at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS, 
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+''' Cozmo looks around and drives after colors.
+
+When the program starts, Cozmo will look around for the color blue.
+Tap the cube illuminated blue to switch the color Cozmo will look for.
+Tap the blinking white cube to have the Tkviewer display Cozmo's pixellated camera view.
+
+
+TODO: determine whether to keep pixel_matrix as is, or convert to numpy matrix
+'''
+
+import asyncio
+import functools
+import math
+import numpy
+import sys
+
+import cozmo
+
+from cozmo.util import degrees, radians, distance_mm, speed_mmps
+from cozmo.lights import Color, Light
+try:
+    from PIL import Image, ImageDraw, ImageColor
+except ImportError:
+    sys.exit('Cannot import from PIL: Do `pip3 install --user Pillow` to install')
+
+''' Color utilities:
+        color_dict (dict): map of color names to regions in color space.
+            Instead of defining a color as a single (R, G, B) point, 
+            colors are defined with a minimum and maximum value for R, G, and B.
+            For example, a point with (R, G, B) = (40, 65, 195) falls in the 'blue' region, 
+            because 0 < 40 < 70, 0 < 65 < 70, and 200 < 195 < 255.
+        
+        map_color_to_light (dict): maps each color name with its cozmo.lights.Light value.
+            Red, green, and blue lights are already defined as constants in lights.py, 
+            but we need to define our own custom Light for yellow.
+
+        color_distance_sqr (fn (color, color_range) => int): returns the squared 
+            euclidean distance of color to color_range. 
+'''
+
+color_dict = {
+'red' : (200, 255, 0, 70, 0, 70), 
+'green' : (0, 70, 200, 255, 0, 70), 
+'blue' : (0, 70, 0, 70, 200, 255), 
+'white' : (200, 255, 200, 255, 200, 255), 
+'black' : (0, 30, 0, 30, 0, 30),
+'yellow' : (200, 255, 200, 255, 0, 70), 
+}
+
+map_color_to_light = {
+'green' : cozmo.lights.green_light, 
+'yellow' : Light(Color(name='yellow', rgb = (255, 255, 0))), 
+'blue' : cozmo.lights.blue_light, 
+'red' : cozmo.lights.red_light
+}
+
+def color_distance_sqr(color, color_range):
```

Comment:
In the docstring, should call out the Args and Returns, like (sample text, not accurate for your method): 

    Args:        
        color (int): blah

    Returns:
        int: The squared euclidean distance of color to color_range